optimum/exporters/onnx/utils.py

```diff
@@ -112,3 +112,34 @@ def get_encoder_decoder_models_for_export(
         models_for_export["decoder_with_past"] = (model, decoder_onnx_config_with_past)
 
     return models_for_export
+
+
+def get_decoder_models_for_export(
+    model: Union["PreTrainedModel", "TFPreTrainedModel"],
+    config: "OnnxConfig",
+) -> Dict[str, Tuple[Union["PreTrainedModel", "TFPreTrainedModel"], "OnnxConfig"]]:
+    """
+    Returns the encoder and decoder parts of the model and their subsequent onnx configs.
+
+    Args:
+        model ([`PreTrainedModel`] or [`TFPreTrainedModel`]):
+            The model to export.
+        config ([`~exporters.onnx.config.OnnxConfig`]):
+            The ONNX configuration associated with the exported model.
+
+    Returns:
+        `Dict[str, Tuple[Union[`PreTrainedModel`, `TFPreTrainedModel`], `OnnxConfig`]: A Dict containing the model and
+        onnx configs for the encoder and decoder parts of the model.
+    """
+    models_for_export = dict()
+
+    models_for_export["decoder"] = (
+        model,
+        config.__class__(model.config, task=config.task, use_past=False, use_present_in_outputs=True),
```

Comment:
I think this is required because if the user passes `causal-lm-with-past` we will fail if we don't explicitly pass `use_past=False` here. This follows: https://github.com/fxmarty/optimum/blob/b7b146944a3810ac72567880dcc4490147ee0b05/optimum/exporters/onnx/utils.py#L105

Am I mistaken?

Reply:
Alright I see, you can also set `use_past_in_inputs=False`.